src/uct/ugni/rdma/ugni_rdma_ep.c

```diff
@@ -243,6 +243,72 @@ static void uct_ugni_amo_unpack64(uct_completion_t *self, ucs_status_t status)
     uct_ugni_invoke_orig_comp(fma, status);
 }
 
+ucs_status_t uct_ugni_ep_atomic_xor64(uct_ep_h tl_ep, uint64_t xor,
```

Comment:
hi
thank you for your PR

all these calls have similar implementation
what about create common call like
```
static inline
ucs_status_t uct_ugni_ep_atomic_op64(uct_ep_h tl_ep, uint64_t and,
                                      uint64_t remote_addr, uct_rkey_t rkey, int op, const char *name)
{
     /* ................. */
    uct_ugni_format_fma_amo(fma, GNI_POST_AMO, op,
                            xor, 0, NULL, remote_addr,
                            rkey, LEN_64, ep, NULL, NULL, NULL);
    ucs_trace_data("Posting AMO %s, GNI_PostFma of size %"PRIx64" value"
                   "%"PRIx64" to %p, with [%"PRIx64" %"PRIx64"]", name,
                   fma->super.desc.length, xor,
                   (void *)fma->super.desc.remote_addr,
                   fma->super.desc.remote_mem_hndl.qword1,
                   fma->super.desc.remote_mem_hndl.qword2);
     /* ................. */
}
```
to minimize code duplication?

Reply:
I was following the structure of the other atomic calls. But I can do that if necessary.

Reply:
@yosefe what do you think?

Reply:
it's up to the ugni maintainer, adding @MattBBaker and @manjugv

Reply:
sounds like a good idea to me. The code will look a bit cleaner / easy to maintain.